src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -135,53 +169,125 @@ trait GraphDAO {
    * @tparam T the type of obj
    * @return the updated vertex
    */
-  def setVertexProperties[T: TypeTag: ClassTag](obj: T, vertex: Vertex): Vertex = {
+  def saveToVertex[T: TypeTag: ClassTag](obj: T, vertex: Vertex, graph: Graph, workspaceName: WorkspaceName): Vertex = {
 
-    def getProperties(obj: T, prefix: String): Iterable[(String, Option[Any])] = {
+    def getProperties(obj: T): Iterable[(String, Option[Any])] = {
       val mirror = ru.runtimeMirror(obj.getClass.getClassLoader)
 
-      for (member <- ru.typeTag[T].tpe.members
-           if (member.asTerm.isVal || member.asTerm.isVar) && member.annotations.exists(_.tree.tpe.typeSymbol == ru.typeOf[VertexProperty@field].typeSymbol)
+      for (member <- ru.typeTag[T].tpe.members if (member.asTerm.isVal || member.asTerm.isVar)
       ) yield {
         val fieldMirror = mirror.reflect(obj).reflectField(member.asTerm)
-        prefix + member.name.decodedName.toString.trim -> (fieldMirror.get match {
+        member.name.decodedName.toString.trim -> (fieldMirror.get match {
           case v: Option[Any] => v
           case v => Option(v)
         })
       }
     }
 
-    getProperties(obj, "_").foreach(_ match {
-      case (key, None) => vertex.removeProperty(key)
+    getProperties(obj).foreach(_ match {
+      case (key, None) =>
+        // can't tell if this is a value or a map anymore so just remove both (there should only be one)
+        vertex.removeProperty(key)
+        vertex.getVertices(Direction.OUT, key).foreach(removeMapVertex)
       case (key, Some(value: DateTime)) => vertex.setProperty(key, value.toDate)
+      case (key, Some(value: Map[_,_])) => serializeAttributeMap(vertex, key, value.asInstanceOf[Map[String, Attribute]], graph, workspaceName)
+      case (key, Some(value: AttributeReference)) => serializeReference(vertex, key, value, graph, workspaceName)
+      case (key, Some(seq: Seq[_])) => seq.headOption match {
+        case None => // empty, do nothing
+        case Some(x: Workflow) =>
+          serializeDomainObjects(vertex, seq.asInstanceOf[Seq[Workflow]], workflowEdge, (w: Workflow) => w.workflowId, graph, workspaceName)
+          val workspaceVertex = getSinglePipelineResult(workspacePipeline(graph, workspaceName.namespace, workspaceName.name)).get
+          new GremlinPipeline(vertex).out(workflowEdge).linkIn(workflowEdge, workspaceVertex).iterate()
```

Comment:
i dont know what this is doing - comments?  what happens if we have a seq of objects that don't match these cases, just won't save it or will exceptions be thrown?